### PR TITLE
Another fix for do cluster #1246

### PR DESCRIPTION
**Pull request by vladbalmos**
I tried your fix but my tests suite fails due to the same issue. I propose moving the publish call in the return promise. That seems to fix it.

### CODE REVIEW

File: initializers/redis.js [hunk 1]
```diff
@@ -111,19 +111,21 @@ await api.redis.publish(payload)
...
       // it is possible for another node to get and work the request before we resolve our write
       // see https://github.com/actionhero/actionhero/issues/1244 for more information
-      let responsePromise
       if (waitForResponse) {
-        responsePromise = new Promise((resolve, reject) => {
+        return new Promise(async (resolve, reject) => {
           let timer = setTimeout(() => reject(new Error('RPC Timeout')), api.config.general.rpcTimeout)
           api.redis.rpcCallbacks[messageId] = {timer, resolve, reject}
+          try {
+            await api.redis.publish(payload)
```
Comment (evantahler) [inline initializers/redis.js]: This will double publish the paylod, no?

File: initializers/redis.js [hunk 2]
```diff
@@ -111,19 +111,21 @@ await api.redis.publish(payload)
...
+          } catch (e) {
+            clearTimeout(timer)
+            delete api.redis.rpcCallbacks[messageId]
+            throw e
+          }
         })
       }
 
       await api.redis.publish(payload)
```
Comment (evantahler) [inline initializers/redis.js]: Should you then remove this publish, or have an `if` around it?

File: initializers/redis.js (hunk 1 above)
Comment (vladbalmos) [reply initializers/redis.js]: If i'm not overlooking something, it shouldn't. If waitForResponse is true, then doCluster returns the promise with the  `try { wait publish } catch (...)` and skips the next publish call.

File: initializers/redis.js (hunk 2 above)
Comment (vladbalmos) [reply initializers/redis.js]: If `waitForResponse` is true, then this `publish` won't be executed

Comment (evantahler) [reply initializers/redis.js]: Ah! I missed the return!

**Pull request merged**